packages/nimble-components/src/rich-text-editor/testing/rich-text-editor.pageobject.ts

```diff
@@ -0,0 +1,65 @@
+import type { RichTextEditor } from '..';
+import type { ToggleButton } from '../../toggle-button';
+
+/**
+ * Page object for the `nimble-rich-text-viewer` component.
+ */
+export class RichTextEditorPageObject {
+    public constructor(
+        private readonly richTextEditorElement: RichTextEditor
+    ) {}
+
+    public getEditorSection(): Element | null | undefined {
+        return this.richTextEditorElement.shadowRoot?.querySelector('.editor');
+    }
+
+    public getTiptapEditor(): Element | null | undefined {
+        return this.richTextEditorElement.shadowRoot?.querySelector(
+            '.ProseMirror'
+        );
+    }
+
+    public getFooterSection(): Element | null | undefined {
+        return this.richTextEditorElement.shadowRoot?.querySelector(
+            '.footer-section'
+        );
+    }
+
+    public getSlotElement(): Element | null | undefined {
+        return this.richTextEditorElement.shadowRoot?.querySelector('slot');
+    }
+
+    public getFormattingButton(className: string): ToggleButton {
+        // eslint-disable-next-line @typescript-eslint/non-nullable-type-assertion-style
```

Comment:
I am returning it as a `ToggleButton | null | undefined`. I hope this will give the test scripts using the page object function to return null and added non-null insertion in the test cases as we did for other cases.